### PR TITLE
Expose const ref of resource and mutex in SharedResource

### PR DESCRIPTION
**Pull request by prashanthr05**
This will allow the proper usage of `SharedResource` with smart pointers across multiple threads.

### CODE REVIEW

File: src/System/include/BipedalLocomotion/System/SharedResource.h [hunk 1]
```diff
@@ -44,6 +46,21 @@ template <class T> class SharedResource
...
+     * Get the resource using const reference.
+     * @note one must not forget to access the mutex and
+     *       perform necessary locking and unlocking
+     * @return the const reference of the object inside the shared resource.
+     */
+    const T& get() const;
+
+    /**
+     * Get the mutex by reference.
+     * @note one must be careful not to modify the mutex
```
Comment (S-Dafarra) [inline src/System/include/BipedalLocomotion/System/SharedResource.h:59]: > std::mutex is neither copyable nor movable.
From https://en.cppreference.com/w/cpp/thread/mutex

So in principle, there is no way to edit it, right?

**Review by S-Dafarra — APPROVED** [review]

File: src/System/include/BipedalLocomotion/System/SharedResource.h (hunk 1 above)
Comment (prashanthr05) [reply src/System/include/BipedalLocomotion/System/SharedResource.h:59]: Great! Then, I will remove that line.

Comment (prashanthr05) [reply src/System/include/BipedalLocomotion/System/SharedResource.h]: Thanks, done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/398/commits/7959a8d53b630076205238f2a2e5917366a5380a.

**Comment by GiulioRomualdi** [comment]
I think this PR is too old and currently not required. I will close for the time being

**Pull request closed without merging**